storage/netapp/sf_check_connections.py

```diff
@@ -0,0 +1,251 @@
+DOCUMENTATION = '''
+
+module: sf_check_connections
+
+short_description: Check connectivity to MVIP and SVIP.
+version_added: '2.3'
+author: Sumit Kumar (sumit4@netapp.com)
+description:
+- Used to test the management connection to the cluster.
+- The test pings the MVIP and SVIP, and executes a simple API method to verify connectivity.
+
+options:
+
+  hostname:
+    required: true
+    description:
+    - hostname
+
+  username:
+    required: true
+    description:
+    - username
+
+  password:
+    required: true
+    description:
+    - password
+
+  skip:
+    required: false
+    description:
+    - Skip checking connection to SVIP or MVIP.
+    choices: ['svip', 'mvip']
+
+  mvip:
+    required: false
+    description:
+    - Optionally, use to test connection of a different MVIP.
+    - This is not needed to test the connection to the target cluster.
+
+  svip:
+    required: false
+    description:
+    - Optionally, use to test connection of a different SVIP.
+    - This is not needed to test the connection to the target cluster.
+
+'''
+
+
+EXAMPLES = """
+   - name: Check connections to MVIP and SVIP
+     sf_check_connections:
+       hostname: "{{ solidfire_hostname }}"
+       username: "{{ solidfire_username }}"
+       password: "{{ solidfire_password }}"
+"""
+
+RETURN = """
+msg:
+    description: Successful
+    returned: success
+    type: string
+    sample: '{"changed": true}'
+
+"""
+
+import sys
+import json
+import logging
+from traceback import format_exc
+
+from ansible.module_utils.basic import *
+from ansible.module_utils.urls import *
+from ansible.module_utils.pycompat24 import get_exception
+
+import socket
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.DEBUG)
+
+logging.basicConfig(level=logging.ERROR)
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.DEBUG)
+
+from solidfire.factory import ElementFactory
+
+
+class SolidFireConnection(object):
+
+    def __init__(self):
+        logger.debug('Init %s', self.__class__.__name__)
+
+        self.module = AnsibleModule(
+            argument_spec=dict(
+                hostname=dict(required=True, type='str'),
+                username=dict(required=True, type='str'),
+                password=dict(required=True, type='str'),
+                skip=dict(required=False, type='str', default=None, choices=['mvip', 'svip']),
+                mvip=dict(required=False, type='str', default=None),
+                svip=dict(required=False, type='str', default=None)
+            ),
+            supports_check_mode=True
+        )
+
+        p = self.module.params
+
+        # set up state variables
+        self.hostname = p['hostname']
+        self.username = p['username']
+        self.password = p['password']
+        self.skip = p['skip']
+        self.mvip = p['mvip']
+        self.svip = p['svip']
+
+        # create connection to solidfire cluster
+        # self.sfe = ElementFactory.create(self.hostname, self.username, self.password)
```

Comment:
You might want to clean this up or at least add the port parameter in the commented code so it matches line 147.